docs/devcontainer-debug.md

```diff
@@ -0,0 +1,134 @@
+```bash
+
+# debug/secrets.env
+
+KEYVAULT_RESOURCE_GROUP=<KEYVAULT_RESOURCE_GROUP>
+KEYVAULT_NAME=<KEYVAULT_NAME>
+KEYVAULT_CLIENT_ID=<KEYVAULT_CLIENT_ID>
+KEYVAULT_CLIENT_SECRET=<KEYVAULT_CLIENT_SECRET>
+AZURE_SUBSCRIPTION_ID=<AZURE_SUBSCRIPTION_ID>
+AZURE_TENANT_ID=<AZURE_TENANT_ID>
+```
+
+### launch.json
+
+  - the `launch.json` file includes an `args` property which has a value of an array of strings. Each string is a command line argument that is needed to be passed to the `secrets-store-csi-driver-provider-azure` executable.
+      - `--attributes` Are the parameters that correspond to the same parameters property in the `SecretProviderClass`. These attributes are used to describe the properties of your Azure Key Vault to get the correct secrets wanted from it (such as the example secrets  `databasePassword` and `storageToken`).
+      - `--secrets` are the credentials from the Service Principal that are needed to access the Azure Key.
+      - `--permission` File permissions. Set to `422`. (422 relates to 644 (Octal) `-rw-r--r--`)
+      - `--debug` set to **true** to allow for debugging of the executable
+
+### parameters.yaml
+
+  - This is a `yaml` file that maps out the attributes for the `secrets-store-csi-driver-provider-azure`.
+  - The `objects` property takes a string that contains a stringed version of an **array of objects.**
+  - **NOTE:** The `objects` property defines which keys should be present within the Key Vault. This is where you would add whichever secrets, credentials, or keys are expected to be accessed within the Azure Key Vault.
+      - Example: If a key named `anExampleKey` was present inside the Azure Key Vault. Another stringed object would be  added inside the `array:` as demonstrated below:
+          ```yaml
+          objects:  |
+            array:
+              - |
+                objectName: databasePassword
+                objectType: secret
+              - |
+                objectName: storageToken
+                objectType: secret
+              - |
+                objectName: anExampleKey
+                objectType: key
+          ```
+
+### build-args .sh
+
+- This file contains a shell script for building the `launch.json`
+- The `SECRETS` env variable pulls the `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET` from the `secrets.env`. These variables are the credentials for authenticating and authorizing usage of the Key Vault.
+- The `ATTRIBUTES` replaces secrets taken from the `secrets.env` with their corresponding key inside the `parameters.yaml`. This env variable contains the properties that explain the nature of the Azure Key Vault such as its name, information on secrets stored inside, tenant_id, resource group, and subscription_id.
+- The command `make build` will build the application and produce the binary `secrets-store-csi-driver-provider-azure` that will be located inside your workspace directory (`secrets-store
```

Comment:
`(`secrets-store` probably should be `(`secrets-store-csi-driver-provider-azure`)?